### PR TITLE
Improvements to JcloudsLocationSecurityGroupCustomizer

### PR DESCRIPTION
**Pull request by sjcorbett**
* Class is no longer final. Useful methods are exposed to subclasses.
* 0.0.0.0/0 used for CIDR for SSH so that Brooklyn failing over doesn't break its SSH access to machines.
* Incorporates SecurityGroupDefinitions.
* Adds ICMP permission to shared group.
* Predicate<Exception> can be provided to indicate that failed requests may be retried.
* Includes AwsExceptionRetryPredicate to retry InvalidGroup.InUse, DependencyViolation and RequestLimitExceeded errors.

### CODE REVIEW

**Comment by sjcorbett** [comment]
Credit to @richardcloudsoft for the first version of the retry logic.

File: locations/jclouds/src/main/java/brooklyn/location/jclouds/networking/JcloudsLocationSecurityGroupCustomizer.java [hunk 1]
```diff
@@ -77,19 +84,31 @@ public JcloudsLocationSecurityGroupCustomizer load(final String appContext) thro
...
+    /**
+     * A predicate indicating whether the customiser can retry a request to add a security group
+     * or a rule after an throwable is thrown.
+     */
+    private Predicate<Exception> isExceptionRetryable = Predicates.alwaysFalse();
+
+    protected JcloudsLocationSecurityGroupCustomizer(String applicationId) {
+        // Would be better to restrict with something like LocalhostExternalIpCidrSupplier, but
+        // we risk making machines inaccessible from Brooklyn when HA fails over.
+        this(applicationId, Suppliers.ofInstance(new Cidr("0.0.0.0/0")));
```
Comment (ahgittin) [inline locations/jclouds/src/main/java/brooklyn/location/jclouds/networking/JcloudsLocationSecurityGroupCustomizer.java]: use `Cidr.UNIVERSAL`

File: locations/jclouds/src/main/java/brooklyn/location/jclouds/networking/JcloudsLocationSecurityGroupCustomizer.java [hunk 2]
```diff
@@ -366,6 +426,62 @@ void clearSecurityGroupCaches() {
     }
 
     /**
+     * Runs the given callable. Repeats until the operation succeeds or {@link #isExceptionRetryable} indicates
+     * that the request cannot be retried.
+     */
+    protected <T> T runOperationWithRetry(Callable<T> operation) {
```
Comment (ahgittin) [inline locations/jclouds/src/main/java/brooklyn/location/jclouds/networking/JcloudsLocationSecurityGroupCustomizer.java]: could use `Repeater` -- it has `backoff` support

**Comment by ahgittin** [comment]
LGTM.  haven't run tests but i believe you that they work.  minor comments, no need for them to block merge.

**Comment by ahgittin** [comment]
@sjcorbett if there are no objections i'm going to merge -- comments are small enough they can be addressed in a new PR

**Comment by sjcorbett** [comment]
@ahgittin Thanks. I will incorporate comments in the future.

**Pull request merged**